components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/InternalAPIDispatcher.java

```diff
@@ -104,6 +107,43 @@ private APIResource findResource(MessageContext synCtx, InternalAPI internalApi)
                     synCtx.setProperty(RESTConstants.REST_URI_VARIABLE_PREFIX + entry.getKey(),
                             entry.getValue());
                 }
+
+                int queryIndex = path.indexOf('?');
+                if (queryIndex != -1) {
```

Comment:
@SanojPunchihewa Since the same code is found in synapse core https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/rest/Resource.java#L322.L356 , Shall we look into the possibility of refactoring that to a method there and using it here